app/models/agents/email_digest_agent.rb

```diff
@@ -29,22 +33,20 @@ def default_options
 
     def receive(incoming_events)
       incoming_events.each do |event|
-        self.memory['queue'] ||= []
-        self.memory['queue'] << event.payload
-        self.memory['events'] ||= []
-        self.memory['events'] << event.id
+        memory['events'] ||= []
+        memory['events'] << event.id
       end
     end
 
     def check
-      if self.memory['queue'] && self.memory['queue'].length > 0
+      if self.memory['events'] && self.memory['events'].length > 0
         ids = self.memory['events'].join(",")
-        groups = self.memory['queue'].map { |payload| present(payload) }
+        events = sort_events(Event.find(memory['events']))
```

Comment:
`Event.where(id: memory['events'])` would be saver, `find` will throw an exception when one of the events was manually deleted.